L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,259 @@
+        public string Key { get; }
+    }
+    public class BalancerState {
+        public BalancerState(ConnectivityState connectivityState, SubChannelPicker picker);
+        public ConnectivityState ConnectivityState { get; }
+        public SubChannelPicker Picker { get; }
+    }
+    public class ChannelState {
+        public ChannelState(AddressResolverResult resolverState, BalancerAttributes options);
+        public BalancerAttributes Options { get; set; }
+        public AddressResolverResult ResolverState { get; set; }
+    }
+    public class CompleteContext {
+        public CompleteContext();
+        public Exception? Error { get; set; }
+        public Metadata? Trailers { get; set; }
+    }
+    public class DnsAddressResolver : AddressResolver {
+        public DnsAddressResolver(Uri address, ILoggerFactory loggerFactory);
+        public override Task RefreshAsync(CancellationToken cancellationToken);
+        public override IDisposable Subscribe(IObserver<AddressResolverResult> observer);
+        protected override void Dispose(bool disposing);
+    }
+    public class DnsAddressResolverFactory : AddressResolverFactory {
+        public DnsAddressResolverFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override AddressResolver Create(Uri address, AddressResolverOptions options);
+    }
+    public abstract class LoadBalancer : IDisposable {
+        protected LoadBalancer();
+        public void Dispose();
+        public abstract void ResolverError(Exception exception);
+        public abstract void UpdateChannelState(ChannelState state);
+        public abstract void UpdateSubChannelState(SubChannel subChannel, SubChannelState state);
+        protected virtual void Dispose(bool disposing);
+    }
+    public abstract class LoadBalancerFactory {
+        protected LoadBalancerFactory();
+        public abstract string Name { get; }
+        public abstract LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class PickContext {
+        public PickContext();
+        public HttpRequestMessage? Request { get; set; }
+        public bool WaitForReady { get; set; }
+    }
+    public class PickFirstBalancer : LoadBalancer {
+        public PickFirstBalancer(IConnectionController channel, ILoggerFactory loggerFactory);
+        public override void ResolverError(Exception exception);
+        public override void UpdateChannelState(ChannelState state);
+        public override void UpdateSubChannelState(SubChannel subChannel, SubChannelState state);
+    }
+    public class PickFirstBalancerFactory : LoadBalancerFactory {
+        public PickFirstBalancerFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class PickResult {
```

Comment:
This should contain the following fields:
- the type of result (COMPLETE, QUEUE, DROP, or FAILED)
- if type is COMPLETE:
  - the subchannel to send the call on
  - an optional callback to intercept the call's trailing metadata (xDS LB policies use this for load reporting data -- this can be added later)
- if type is DROP or FAILED:
  - a status code and message to fail the RPC with

Reply:
What is drop used for? Is it to cause an RPC to fail immediately even when wait for ready is true?

Reply:
Yes, and it also inhibits retries.